src/libraries/System.Private.CoreLib/src/System/TimeZoneInfo.Unix.Android.cs

```diff
@@ -142,6 +142,56 @@ private static TimeZoneInfo GetLocalTimeZoneCore()
             return Utc;
         }
 
+        private static TimeSpan GetCacheLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+                CachedData cachedData = s_cachedData;
+                return cachedData.Local.GetUtcOffset(dateTime, flags, cachedData);
+        }
+
+        private static TimeSpan? _localUtcOffset;
+        private static object _localUtcOffsetLock = new();
+        private static Thread? _loadAndroidTZData;
+        // Shortcut for TimeZoneInfo.Local.GetUtcOffset
+        // On Android, loading AndroidTZData while obtaining cachedData.Local is expensive for startup.
+        // We introduce a fast result for GetLocalUtcOffset that relies on the date time offset being
+        // passed into monovm_initialize(_preparsed) from Java in seconds as LOCAL_DATE_TIME_OFFSET.
+        // However, to handle timezone changes during the app lifetime, AndroidTZData needs to be loaded.
+        // The fast path is initially used, and we start a background thread to get cachedData.Local
+        internal static TimeSpan GetLocalUtcOffset(DateTime dateTime, TimeZoneInfoOptions flags)
+        {
+            if (_localUtcOffset != null) // The background thread finished, the cache is loaded.
+                return GetCacheLocalUtcOffset(dateTime, flags);
+
+            if (_localUtcOffset == null && _loadAndroidTZData == null) // The cache isn't loaded and no background thread has been created
+            {
+                lock (_localUtcOffsetLock)
+                {
+                    // GetLocalUtcOffset may be called multiple times before a cache is loaded and a background thread is running,
+                    // once the lock is available, check for a cache and background thread.
+                    if (_localUtcOffset != null)
+                        return GetCacheLocalUtcOffset(dateTime, flags);
+
+                    if (_loadAndroidTZData == null)
+                    {
+                        _loadAndroidTZData = new Thread(() => {
+                            Thread.Sleep(1000);
+                            _localUtcOffset = GetCacheLocalUtcOffset(dateTime, flags);
+                        });
+                        _loadAndroidTZData.IsBackground = true;
+                        _loadAndroidTZData.Start();
```

Comment:
Who is going to remove/clear the thread?

Reply:
Whats the best way to do that? Is the way to remove/clear Threads to set `_loadAndroidTZData` to `null` once we know the thread is finished? (maybe in the first `if (_localUtcOffset != null) // The background thread finished, the cache is loaded.`)

Reply:
Setting `s_loadAndroidTZData` to null when we know AndroidTZData is loaded (`s_androidTZDataLoaded == true`)